specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+                    // You can ask the user to provide information like file name, file extension, and so on. 
+                    // Finally, and set them on the event args
+                    //
+                    // This is a customized dialog example, the constructor returns after the 
+                    // dialog interaction is completed by the end user.
+                    SaveAsDialog dialog(m_appWindow->GetMainWindow(), saveKinds);
+                    if (dialog.confirmed)
+                    {
+                        // Setting the ResultFilePath, Kind, AllowReplace for the event
+                        // args from this customized dialog inputs is optional.
+                        // The event args has default values based on the document to save.
+                        CHECK_FAILURE(
+                            args->put_ResultFilePath((LPCWSTR)dialog.path.c_str()));
+                        CHECK_FAILURE(args->put_Kind(dialog.selectedKind));
+                        CHECK_FAILURE(args->put_AllowReplace(dialog.allowReplace));
+                    }
+                    else
+                    {
+                        // Save As cancelled from this customized dialog
+                        CHECK_FAILURE(args->put_Cancel(TRUE));
+                    }
+                };
+
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                m_appWindow->RunAsync(
+                    [deferral, showCustomizedDialog]()
+                    {
+                        showCustomizedDialog();
+                        CHECK_FAILURE(deferral->Complete());
+                    });
+                return S_OK;
+            })
+            .Get(),
+        &m_saveAsRequestedToken);
+    
+    MessageBox(
+        m_appWindow->GetMainWindow(), L"Event Handler Added", L"Info",MB_OK);
+    return true;
+}
+```
+### Programmatic Save As
+Call RequestSaveAs method to trigger the programmatic save as.
+```c++
+
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2_20)
+        return false;
+    m_webView2_20->RequestSaveAs(
+        Callback<ICoreWebView2RequestSaveAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS result) -> HRESULT
+            {
+                // Show RequestSaveAs returned result, optional
+                MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+```
+
+## .Net/ WinRT
+### Add the Event Handler
+This example hides the default save as dialog and shows a customized dialog.
+```c#
+
+void AddEventHandlerExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.SaveAsRequested -= WebView_SaveAsRequested;
```

Comment:
Why does this sample want to remove its event handler after it runs once?  Is the context that his was registered for a one-time handling of programmatic SaveAs but I don't want to handle user-initiated save actions?

Reply:
Please see above.